lib/auth/mocku2f/webauthn.go

```diff
@@ -0,0 +1,82 @@
+package mocku2f
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/duo-labs/webauthn/protocol"
+	"github.com/gravitational/teleport/lib/auth/webauthn"
+	"github.com/gravitational/trace"
+)
+
+// collectedClientData is part of the data signed by authenticators (after
+// marshaled to JSON, hashed and appended to authData).
+// https://www.w3.org/TR/webauthn-2/#dictionary-client-data
+type collectedClientData struct {
+	Type      string `json:"type"`
+	Challenge string `json:"challenge"`
+	Origin    string `json:"origin"`
+}
+
+// SignAssertion signs a WebAuthn assertion following the
+// U2F-compat-getAssertion algorithm.
+func (muk *Key) SignAssertion(origin string, assertion *webauthn.CredentialAssertion) (*webauthn.CredentialAssertionResponse, error) {
+	// Is our credential allowed?
+	ok := false
+	for _, c := range assertion.Response.AllowedCredentials {
+		if bytes.Equal(c.CredentialID, muk.KeyHandle) {
+			ok = true
+			break
+		}
+	}
+	if !ok {
+		return nil, trace.Errorf("device not allowed")
+	}
+
+	// Is the U2F app ID present?
+	appID := assertion.Response.Extensions["appid"].(string)
```

Comment:
We can probably store this string as a constant in `constants.go` and replace all "appid" strings with that.

Reply:
Done, but as a public constant inside lib/auth/webauthn (instead of package teleport/constants.go).